app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt

```diff
@@ -578,6 +578,7 @@ abstract class BaseBrowserFragment : Fragment(), UserInteractionHandler, Session
             val sessionManager = requireComponents.core.sessionManager
             if (session.source == Session.Source.ACTION_VIEW) {
                 sessionManager.remove(session)
+                activity!!.onBackPressed()
```

Comment:
Yeah, okay! I'm down with trying this approach and see if any weird edge cases that I can't think of right now come up.
Would you be okay with this change? 
```suggestion
                activity?.onBackPressed()
```